################################################################################
SHORT DESCRIPTION:
################################################################################
Handle the xci cpuid signature in LibXL

################################################################################
LONG DESCRIPTION:
################################################################################
This is a hack from a long time ago, when we had to use custom pv frontends.
I is now disabled by default but still very useful for debugging.
A seperate patch introduces the feature as a hypercall, this one just adds
support for it in XL VM configuration files.

################################################################################
CHANGELOG
################################################################################
Authors:
Jed Lejosne <lejosnej@ainfosec.com>

################################################################################
REMOVAL
################################################################################
Removing this won't break anything, but disable a useful debugging feature.

################################################################################
UPSTREAM PLAN
################################################################################
This is specific to OpenXT, nothing to upstream here.

################################################################################
INTERNAL DEPENDENCIES
################################################################################
openxt-xci-cpuid-signature.patch

################################################################################
PATCHES
################################################################################
Index: xen-4.10.0/tools/libxl/libxl_create.c
===================================================================
--- xen-4.10.0.orig/tools/libxl/libxl_create.c
+++ xen-4.10.0/tools/libxl/libxl_create.c
@@ -330,6 +330,7 @@ int libxl__domain_build_info_setdefault(
         libxl_defbool_setdefault(&b_info->u.hvm.hpet,               true);
         libxl_defbool_setdefault(&b_info->u.hvm.vpt_align,          true);
         libxl_defbool_setdefault(&b_info->u.hvm.altp2m,             false);
+        libxl_defbool_setdefault(&b_info->u.hvm.xci_cpuid_signature, false);
         libxl_defbool_setdefault(&b_info->u.hvm.usb,                false);
         libxl_defbool_setdefault(&b_info->u.hvm.xen_platform_pci,   true);
 
Index: xen-4.10.0/tools/libxl/libxl_types.idl
===================================================================
--- xen-4.10.0.orig/tools/libxl/libxl_types.idl
+++ xen-4.10.0/tools/libxl/libxl_types.idl
@@ -558,6 +558,7 @@ libxl_domain_build_info = Struct("domain
                                        # for x86 HVM guests and is maintained
                                        # for legacy purposes.
                                        ("altp2m",           libxl_defbool),
+                                       ("xci_cpuid_signature", libxl_defbool),
                                        ("system_firmware",  string),
                                        ("smbios_firmware",  string),
                                        ("acpi_firmware",    string),
Index: xen-4.10.0/tools/xl/xl_parse.c
===================================================================
--- xen-4.10.0.orig/tools/xl/xl_parse.c
+++ xen-4.10.0/tools/xl/xl_parse.c
@@ -1320,6 +1320,8 @@ void parse_config_data(const char *confi
             fprintf(stderr, "WARNING: Specifying \"altp2mhvm\" is deprecated. "
                     "Please use \"altp2m\" instead.\n");
 
+        xlu_cfg_get_defbool(config, "xci_cpuid_signature", &b_info->u.hvm.xci_cpuid_signature, 0);
+
         xlu_cfg_replace_string(config, "smbios_firmware",
                                &b_info->u.hvm.smbios_firmware, 0);
         xlu_cfg_replace_string(config, "acpi_firmware",
Index: xen-4.10.0/tools/libxl/libxl_dom.c
===================================================================
--- xen-4.10.0.orig/tools/libxl/libxl_dom.c
+++ xen-4.10.0/tools/libxl/libxl_dom.c
@@ -338,6 +338,9 @@ static void hvm_set_conf_params(xc_inter
                          libxl_defbool_val(info->u.hvm.vpt_align));
         xc_hvm_param_set(handle, domid, HVM_PARAM_NESTEDHVM,
                          libxl_defbool_val(info->nested_hvm));
+        xc_hvm_param_set(handle, domid, HVM_PARAM_XCI_CPUID_SIGNATURE,
+                         libxl_defbool_val(info->u.hvm.xci_cpuid_signature));
+
         break;
     default:
         abort();
